Apps/DiceRoller/DiceRoller/ViewController.m

```diff
@@ -30,22 +52,289 @@ - (void)didReceiveMemoryWarning{
+        
+        //NSString *userEmail = @"mobile.fake@example.com";
+        //NSString *password = @"mobile";
+        
+        // login to the iSENSE site
+        [api createSessionWithEmail:userEmail andPassword:password];
+        
+        // pull down fields for the iSENSE project as an array of RProjectField objects
+        dm = [DataManager getInstance];
+        NSArray *projectFields = [api getProjectFieldsWithId:[dm getProjectID]];
+        
+        // initialize an array to store your data
+        NSMutableArray *rowMajorData = [[NSMutableArray alloc] init];
+        
+        // each data row is a dictionary, where the keys are the project's field IDs and the values
+        // are the data you want to store for each field.  if you want more data rows per data
+        // set, you can declare additional dictionaries, populate them as we will below, and also add
+        // them to the data array.  PLEASE NOTE that iSENSE only accepts strings as keys and values,
+        // however, so when adding the data to the dictionary, store the field ID and value as a string
+        NSMutableDictionary *dataRow = [[NSMutableDictionary alloc] init];
+        
+        // loop through the project fields and store each data point in a dictionary
+        for (RProjectField *field in projectFields) {
+            
+            // see if this field contains "white" or "yellow" in it
+            if ([field.name.lowercaseString rangeOfString:@"white"].location != NSNotFound) {
+                
+                // if we're here, this is our value of the white dice
+                // remember, the object and key should be strings!
+                [dataRow setObject:[NSString stringWithFormat:@"%d", whiteDiceValue]
+                            forKey:[NSString stringWithFormat:@"%d", field.field_id.intValue]];
+                
+                
+            } else if ([field.name.lowercaseString rangeOfString:@"yellow"].location != NSNotFound) {
+                
+                // if we're here, this is our value of the yellow dice
+                [dataRow setObject:[NSString stringWithFormat:@"%d", yellowDiceValue]
+                            forKey:[NSString stringWithFormat:@"%d", field.field_id.intValue]];
+                
+            } else {
+                
+                // if we're here, this is the dice sum value
+                [dataRow setObject:[NSString stringWithFormat:@"%d", diceRollSum]
+                            forKey:[NSString stringWithFormat:@"%d", field.field_id.intValue]];
+                
+            }
+        }
+        
+        // add this data row to our data array
+        [rowMajorData addObject:dataRow];
+        
+        // at this point, feel free to add more data points to the rowMajorData array!
+        // to prevent the need for parsing the project's fields and logging in multiple times,
+        // make sure you only make these API calls once though.
```

Comment:
This set of comments can be deleted:
```
+        // at this point, feel free to add more data points to the rowMajorData array!
+        // to prevent the need for parsing the project's fields and logging in multiple times,
+        // make sure you only make these API calls once though.
```